travis_ci.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+set -e # fast fail
+
+# action for travis ci
+Action=$1
+# config the number of integration test job.
+JOBS=$2
+# config the id of job
+JOB_ID=$3
+
+CASE_PREFIX="testcase.list"
+LOCAL_CASE_FILE="${TRAVIS_BUILD_DIR}/test/${CASE_PREFIX}"
+
+source "${TRAVIS_BUILD_DIR}/hack/install/config.sh"
+
+prepare_testcase() {
+  # get the number of all test case
+  # test case must write with this format:
+  # func (XXX) TestXXX(c *check.C) {
+  grep -h -E "^func\ \(.*\)\ Test.*\(c\ \*check\.C\)\ \{" ./test -r \
+    | awk '{print $3,$4}' | awk -F \( '{print $1}' \
+    | sed 's/\*//' | sed 's/) /./' > "${LOCAL_CASE_FILE}"
+
+  # make test case list file
+  local sum nums
+  local index=1
+  local loop=1
+
+  # shellcheck disable=SC2002
+  sum=$(cat "${LOCAL_CASE_FILE}" | wc -l)
+  nums=$((sum / JOBS + 1))
+
+  rm -rf "${LOCAL_CASE_FILE}.*"
+  # shellcheck disable=SC2013
+  for test in $(cat "${LOCAL_CASE_FILE}"); do
```

Comment:
using `split` command. `split -l ${nums} -d --suffix-length=1 ${LOCAL_CASE_FILE} ${LOCAL_CASE_FILE}.`

an example 
```bash
[root@pouch test]# ls
example
[root@pouch test]# cat example
1
2
33
44
5
6
7
[root@pouch test]# split -l 3 -d --suffix-length=1 example name.
[root@pouch test]# ls
example  name.0  name.1  name.2
[root@pouch test]# cat name.0
1
2
33
[root@pouch test]# cat name.1
44
5
6
[root@pouch test]# cat name.2
7
```

Reply:
wow, good idea. thank you for your suggestion.